s5_base/run.sh

```diff
@@ -6,148 +6,284 @@
+    2500 15000 \
+    data/train_40k data/lang \
+    exp/tri1_ali exp/tri2
+    
 fi
 
-# train an LDA+MLLT system triphone2.
-if [ $stage -le 7 ]; then
-  steps/train_lda_mllt.sh --cmd "$train_cmd" \
-    --splice-opts "--left-context=3 --right-context=3" 2500 15000 \
-    data/train data/lang exp/tri1_ali_train exp/samromur_tri2b
-
-  # decode using the LDA+MLLT model
-  (
-    utils/mkgraph.sh data/lang_3gsmall \
-      exp/samromur_tri2b exp/samromur_tri2b/graph_3gsmall
-    for test in dev; do
-      steps/decode.sh --nj 10 --cmd "$decode_cmd" exp/samromur_tri2b/graph_3gsmall \
-        data/$test exp/samromur_tri2b/decode_3gsmall_$test
-      steps/lmrescore.sh --cmd "$decode_cmd" data/lang_{3gsmall,3gmed} \
-        data/$test exp/samromur_tri2b/decode_{3gsmall,3gmed}_$test
-      steps/lmrescore_const_arpa.sh \
-        --cmd "$decode_cmd" data/lang_{3gsmall,3glarge} \
-        data/$test exp/samromur_tri2b/decode_{3gsmall,3glarge}_$test
+if [ $stage -le 10 ]; then
+    echo "Second triphone decoding"
+    $train_cmd --mem 4G exp/tri2/log/mkgraph.log \
+    utils/mkgraph.sh data/lang_3g exp/tri2 exp/tri2/graph
+    
+    for dir in dev eval; do
+        (
+            steps/decode.sh \
+            --config conf/decode.config \
+            --nj "$nj_decode" --cmd "$decode_cmd" \
+            exp/tri2/graph data/$dir \
+            exp/tri2/decode_$dir;
+            
+            steps/lmrescore_const_arpa.sh \
+            --cmd "$decode_cmd" \
+            data/lang_{3g,4g} data/$dir \
+            exp/tri2/decode_$dir \
+            exp/tri2/decode_${dir}_rescored
+        ) &
     done
-  )&
+    wait
+    
+fi
+
+if [ $stage -le 11 ]; then
+    echo "Aligning the full training set to tri2"
+    steps/align_si.sh \
+    --nj $num_jobs --cmd "$train_cmd" \
+    data/train data/lang \
+    exp/tri2 exp/tri2_ali
+    
+    echo "Train LDA + MLLT + SAT"
+    steps/train_sat.sh    \
+    --cmd "$train_cmd --mem 4G" \
+    4000 40000    \
+    data/train data/lang     \
+    exp/tri2_ali exp/tri3
+    
+fi
 
-  # Align utts using the tri2b model
-  steps/align_si.sh  --nj 5 --cmd "$train_cmd" --use-graphs true \
-    data/train data/lang exp/samromur_tri2b exp/samromur_tri2b_ali_train
+if [ $stage -le 12 ]; then
+    echo "Third triphone decoding"
+    $train_cmd --mem 4G exp/tri3/log/mkgraph.log \
+    utils/mkgraph.sh data/lang_3g exp/tri3 exp/tri3/graph
+    
+    for dir in dev eval; do
+        (
+            steps/decode_fmllr.sh \
+            --config conf/decode.config \
+            --nj "$nj_decode" --cmd "$decode_cmd" \
+            exp/tri3/graph data/$dir \
+            exp/tri3/decode_$dir;
+            
+            steps/lmrescore_const_arpa.sh \
+            --cmd "$decode_cmd" \
+            data/lang_{3g,4g} data/$dir \
+            exp/tri3/decode_$dir \
+            exp/tri3/decode_${dir}_rescored
+        ) &
+    done
+    wait
 fi
 
+# WER info:
+mkdir -p RESULTS
```

Comment:
since RESULTS isn't a dir, wouldn't touch make more sense?
```suggestion
touch RESULTS
```